packages/kbn-expandable-flyout/src/hooks/use_expandable_flyout_api.ts

```diff
@@ -25,6 +26,9 @@ import { FlyoutPanelProps, type ExpandableFlyoutApi } from '../types';
 
 export type { ExpandableFlyoutApi };
 
+// observable to notify when the flyout is closed
+const onCloseObs$ = new Subject<boolean>();
```

Comment:
so the same observable is used for all the flyouts currently rendered on the screen?

Reply:
yup, and I think you just found a problem...